src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/CSharpInlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -424,38 +363,60 @@ private static SyntaxNode RemoveDeclaratorFromScope(VariableDeclaratorSyntax var
             return newScope.RemoveNode(newLocalDeclaration, SyntaxRemoveOptions.KeepNoTrivia);
         }
 
-        private static ExpressionSyntax SkipRedundantExteriorParentheses(ExpressionSyntax expression)
+        private static async Task<ExpressionSyntax> CreateExpressionToInlineAsync(
```

Comment:
diff is terrible, but this code is almsot the same.  Generally the logic is similar to: if we have an initializer like `var = new X()` leave it alone (we know the type of it). If we have `X x = new()` change that to `new X()`.  Otherwise, insert an explicit cast from the expression to the final local type.  This ensures the expr keeps the same meaning and affects code the same way in all the places it is inlined to.